docs/_includes/ds6/dialog.html

```diff
@@ -0,0 +1,489 @@
+    <button aria-label="Close dialog" class="dialog__close" type="button">
+      <svg aria-hidden="true" focusable="false" height="17" width="17">
+        <use xlink:href="#icon-close"></use>
+      </svg>
+    </button>
+    <div class="dialog__body">
+      <h2>Heading</h2>
+      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
+        magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+        consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
+        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
+      <p>
+        <a href="http://www.ebay.com">www.ebay.com</a>
+      </p>
+    </div>
+  </div>
+</div>
+    {% endhighlight %}
+  </div>
+
+  <h3 id="dialog-transitions">Dialog Transitions</h3>
+  <p>Skin currently supports two types of dialog transition: fade and slide.
+  Because CSS cannot transition an element to and from <span class="highlight">hidden</span> (i.e. "display:none"), transitions are acheived using two classes applied during different stages of the animation.
+  Before applying an animation class to the dialog component you must first apply the primer class which will be <span class="highlight">$name--from</span> where <span class="highlight">$name</span> is the base class.
+  The dialog component has two different animation base classes <span class="highlight">dialog-show</span> and <span class="highlight">dialog-hide</span>.
+  </p>
+  <p>Firstly the <span class="highlight">--from</span> modifier must be applied to dialog component to prime animation.
+  One animation frame later the <span class="highlight">--from</span> modifier must be removed and the base class applied to start the animation.
+  Finally once the animation has completed remove the base class.
+  </p>
+
+  <p>An example implementation is shown below.</p>
+
+  {% highlight js %}
+const $dialog = document.querySelector(".dialog");
+const $window = $dialog.querySelector(".dialog__window");
+const $openBtn = document.querySelector("#dialog-open");
+const $closeBtn = $dialog.querySelector(".dialog__close");
+
+// Trigger dialog show animation.
+$openBtn.addEventListener("click", () => {
+  $dialog.classList.add("dialog-show--from");
+  $dialog.removeAttribute("hidden");
+
+  requestAnimationFrame(() => requestAnimationFrame(() => {
+    // Two RAFS to ensure this happens on the next animation frame.
+    $dialog.classList.remove("dialog-show--from");
+    $dialog.classList.add("dialog-show");
+
+    $window.addEventListener("transitionend", () => {
+      /**
+       * The window animation is the longest, so we wait for it to
+       * finish before removing the classes.
+       */
+      $dialog.classList.remove("dialog-show");
+    }, { once: true });
```

Comment:
Caniuse here: https://caniuse.com/#feat=once-event-listener

We are not actually using this anywhere but I thought that it helped keep the examples clean rather than manually removing the listener.

Reply:
Ah, I see. That's fine then.